pubtools/_quay/hooks.py

```diff
@@ -47,4 +47,13 @@ def quay_images_untagged(untag_refs, lost_refs):
     """
 
 
+@hookspec
+def get_cert_key_paths_plugin(server_url):
```

Comment:
In pub. The function is 'get_cert_key_paths', which is used in multiple places in the code. I thought using hooks would be a good way to prevent code duplication

Reply:
I decided to go the other direction and remove and reference to Pyxis. The function in pub is generic and should work for any service, so I decided that pubtools-quay's documentation should reflect that, even though the function only used for Pyxis right now.

Reply:
If you want to position this as a generic hook, can you please add it to `pubtools` instead?

It doesn't seem like a good idea to have a hook which is intended to be shared (i.e. hook possibly invoked from outside of pubtools-quay), but the hookspec lives in pubtools-quay project.  I expect it risks some problems such as projects working differently depending on whether pubtools-quay was installed & imported.  At the very least it gives pubtools-quay the ability to break all the projects using it by making backwards-incompatible changes to the hookspec.

I'm warming up to the idea of having this as a generic hook in pubtools... if you do want to go through with that, may I suggest some other minor changes:

- should it have 'firstresult' in the hookspec?
- there are many possible formats of cert/key, so it'd probably be worthwhile for the doc to clarify that we're expecting PEM-formatted files ("BEGIN CERTIFICATE" and "BEGIN RSA PRIVATE KEY") and maybe also that it's permitted for both paths to be the same.
- the fallback strategy part should probably be changed to not mention target settings.  If it's generic, it doesn't make sense to say "...and here's what pubtools-quay does if this hook isn't implemented".  I think the doc string should be written with the primary target audience being whoever wants to invoke the hook, so from their point of view, the relevant information is something like: "Callers of this hook should be prepared to receive no result, and should implement a reasonable fallback strategy in that case."

Reply:
Created a PR for pubtools. I think this PR is blocked until a new pubtools release, not sure how to test the hook otherwise.